tensorflow-utils/src/main/java/org/tensorflow/util/ndarray/NdArraySequence.java

```diff
@@ -0,0 +1,29 @@
+package org.tensorflow.util.ndarray;
+
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+/**
+ * Iterates through a sequence of elements of an N-dimensional array.
+ *
+ * @param <T> data type of the array being iterated
+ */
+public interface NdArraySequence<T extends NdArray<?>> {
```

Comment:
Can this be an iterator? Or implement Iterable?

Reply:
I think that *now*, it should (as I changed its implementation many times...). I'll update it if that's the case.